vibi-dpu/src/core/review.rs

```diff
@@ -94,41 +98,51 @@ async fn commit_check(review: &Review) {
-	println!("data == {:?}", &data["eventPayload"]["repository"]);
-	let repo_provider = data["repositoryProvider"].to_string().trim_matches('"').to_string();
-	let repo_name = data["eventPayload"]["repository"]["name"].to_string().trim_matches('"').to_string();
+	let deserialized_data = data_res.expect("Uncaught error in deserializing message_data");
+	println!("deserialized_data == {:?}", &deserialized_data["eventPayload"]["repository"]);
+	let repo_provider = deserialized_data["repositoryProvider"].to_string().trim_matches('"').to_string();
+	let repo_name = deserialized_data["eventPayload"]["repository"]["name"].to_string().trim_matches('"').to_string();
 	println!("repo NAME == {}", &repo_name);
-	let workspace_name = data["eventPayload"]["repository"]["workspace"]["slug"].to_string().trim_matches('"').to_string();
+	let workspace_name = deserialized_data["eventPayload"]["repository"]["workspace"]["slug"].to_string().trim_matches('"').to_string();
 	let clone_opt = get_clone_url_clone_dir(&repo_provider, &workspace_name, &repo_name);
 	if clone_opt.is_none() {
 		eprintln!("Unable to get clone url and directory");
 		return None;
 	}
 	let (clone_url, clone_dir) = clone_opt.expect("Empty clone_opt");
-	let pr_id = data["eventPayload"]["pullrequest"]["id"].to_string().trim_matches('"').to_string();
+	let pr_id = deserialized_data["eventPayload"]["pullrequest"]["id"].to_string().trim_matches('"').to_string();
 	let review = Review::new(
-		data["eventPayload"]["pullrequest"]["destination"]["commit"]["hash"].to_string().replace("\"", ""),
-		data["eventPayload"]["pullrequest"]["source"]["commit"]["hash"].to_string().replace("\"", ""),
+		deserialized_data["eventPayload"]["pullrequest"]["destination"]["commit"]["hash"].to_string().replace("\"", ""),
+		deserialized_data["eventPayload"]["pullrequest"]["source"]["commit"]["hash"].to_string().replace("\"", ""),
 		pr_id.clone(),
 		repo_name.clone(),
 		workspace_name.clone(),
 		repo_provider.clone(),
 		format!("bitbucket/{}/{}/{}", &workspace_name, &repo_name, &pr_id),
 		clone_dir,
 		clone_url,
-		data["eventPayload"]["pullrequest"]["author"]["uuid"].to_string().replace("\"", ""),
+		deserialized_data["eventPayload"]["pullrequest"]["author"]["uuid"].to_string().replace("\"", ""),
 	);
 	println!("review = {:?}", &review);
 	save_review_to_db(&review);
-	return Some(review);
+	let repo_config_res = serde_json::from_value(deserialized_data["repoConfig"].clone());
+	if repo_config_res.is_err() {
+		let e = repo_config_res.expect_err("No error in repo_config_res");
+		eprintln!("Unable to deserialze repo_config_res: {:?}", e);
+		let default_config = RepoConfig::default();
+		return Some((review, default_config));
+	}
+	let repo_config = repo_config_res.expect("Uncaught error in repo_config_res");
+	println!("repo_config = {:?}", &repo_config);
+	save_repo_config_to_db(&repo_config, &review.repo_name(), &review.repo_owner(), &review.provider());
```

Comment:
We are saving this in the db, but are we ever using this value from the db?

I think we should use it as the 1st default value in this function itself if config was not sent in the `message_data`; and default to the `RepoConfig::default()` only when this information is not available in the db.

Reply:
Agreed, making this an issue, will resolve in another PR